File: 2087. Minimum Cost Homecoming of a Robot in a Grid/Solution.cpp
#include <numeric>
#include <vector>
using namespace std;

class Solution {
public:
    int minCost(vector<int>& startPos, vector<int>& homePos, vector<int>& rowCosts, vector<int>& colCosts)
    {
        int r0 = startPos[0], c0 = startPos[1];
        int r1 = homePos[0], c1 = homePos[1];
        int result = -rowCosts[r0] - colCosts[c0];
        result += reduce(rowCosts.begin() + min(r0, r1), rowCosts.begin() + max(r0, r1) + 1, 0);
        result += reduce(colCosts.begin() + min(c0, c1), colCosts.begin() + max(c0, c1) + 1, 0);
        return result;
    }
};

File: 2087. Minimum Cost Homecoming of a Robot in a Grid/Solution.py
class Solution:
    def minCost(self, startPos: list[int], homePos: list[int], rowCosts: list[int], colCosts: list[int]) -> int:
        r0, c0 = startPos
        r1, c1 = homePos
        result = -rowCosts[r0] - colCosts[c0]
        result += sum(rowCosts[min(r0, r1) : max(r0, r1) + 1])
        result += sum(colCosts[min(c0, c1) : max(c0, c1) + 1])
        return result
